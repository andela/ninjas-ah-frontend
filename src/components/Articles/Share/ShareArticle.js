import React from 'react';
import {
  FacebookShareButton,
  FacebookIcon,
  TwitterShareButton,
  TwitterIcon,
  LinkedinShareButton,
  LinkedinIcon,
  EmailShareButton,
  EmailIcon
} from 'react-share';

const ShareArticle = () => {
  const shareArticle = window.location.href;
  return (
    <div>
      <div>
        <FacebookShareButton className="small-h-margin left cursor-pointer" url={shareArticle}>
          <FacebookIcon size={32} round black />
        </FacebookShareButton>
        <TwitterShareButton className="small-h-margin left cursor-pointer" url={shareArticle}>
          <TwitterIcon size={32} round />
        </TwitterShareButton>
        <LinkedinShareButton
          className="small-h-margin left cursor-pointer"
          url={shareArticle}
          windowWidth={750}
          windowHeight={600}
        >
          <LinkedinIcon size={32} round />
        </LinkedinShareButton>
        <EmailShareButton
          className="small-h-margin left cursor-pointer"
          url={shareArticle}
          body="body"
        >
          <EmailIcon size={32} round />
        </EmailShareButton>
      </div>
      <div className="divider" />
    </div>
  );
};

export default ShareArticle;
