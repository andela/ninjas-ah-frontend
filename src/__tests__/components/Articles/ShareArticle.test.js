import React from 'react';
import { shallow } from '../../../../config/enzymeConfig';

import ShareArticle from '../../../components/Articles/Share/ShareArticle';

describe('<ShareArticle />', () => {
  const component = shallow(<ShareArticle />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });
});
