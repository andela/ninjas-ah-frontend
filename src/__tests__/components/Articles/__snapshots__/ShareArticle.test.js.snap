// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShareArticle /> should render without crashing 1`] = `ShallowWrapper {}`;
